File: react_native/HFilmTrainingApp/src/features/film/film-reducers.js
// import {actionTypes} from "./FilmActions";

export const actionTypes = {
  ADD_FILM: 'ADD_FILM',
  CLICK_FILM: 'CLICK_FILM',
  UNCLICK_FILM: 'CLICK_FILM',
  CLICK_LIKE: 'CLICK_LIKE',
};

export const addFilm = films => ({
  type: actionTypes.ADD_FILM,
  films
});

export const clickLike = id => ({
  type: actionTypes.CLICK_LIKE,
  id
});


const FilmReducers = (films = [], action) => {
  switch (action.type) {
    case actionTypes.ADD_FILM:
      console.log("add_film");
      return [...films, ...action.films];

    case actionTypes.CLICK_LIKE:
      return films.map(film =>
        (film.id === action.id) ?
          {...film, like: !film.like} : film);

    default:
      return films;
  }
};

export default FilmReducers

File: HFilmTraining/lib/features/film/film_reducers.dart

import 'package:hfilmtrainingapp/app_state.dart';
import 'package:hfilmtrainingapp/datas/film/Film.dart';

class AddFilmsAction {
  final List<Film> films;

  AddFilmsAction(this.films);
}

class LikeFilmAction {
  final Film film;
  LikeFilmAction(this.film);
}

class FetchFilmsAction {
}


AppState filmReducers (AppState state, dynamic action) {

  if (action is AddFilmsAction) {
    return new AppState(films: List.from(state.films)..addAll(action.films));

  } else if (action is LikeFilmAction) {
    int id = action.film.id;
    return new AppState(films: state.films.map((film) {
      if (film.id == id) {
        film.like = !film.like;
        print(film.id);
      }
      return film;
    }).toList());
  }

  return state;

}